packages/yoga/src/Drawer/web/Header.jsx

```diff
@@ -1,9 +1,100 @@
 import React from 'react';
+import { bool, string, func } from 'prop-types';
+import { Close } from '@gympass/yoga-icons';
 import Box from '../../Box';
+import { Divider, Heading } from '../..';
 
-const Header = props => (
-  <Box as="header" width="100%" color="text.primary" {...props} />
-);
+function Header({ onClose, title, backHandler, divider, hideCloseButton }) {
+  function showCloseButton() {
+    if (onClose && !hideCloseButton) {
+      return (
+        <Heading.RightButton
+          aria-label="close-button-drawer"
+          onClick={onClose}
+          icon={Close}
+          large
+        />
+      );
+    }
+
+    return null;
+  }
+
+  function showTitle() {
+    if (title) {
+      return (
+        <Heading.Title
+          style={{
+            letterSpacing: '1px',
+            textTransform: 'uppercase',
+            fontSize: '12px',
+          }}
+        >
+          {title}
+        </Heading.Title>
+      );
+    }
+
+    return null;
+  }
+
+  function showBackButton() {
+    if (backHandler) {
+      return (
+        <Heading.BackButton
+          aria-label="back-button-drawer"
+          onClick={backHandler}
+        />
+      );
+    }
+
+    return null;
+  }
+
+  function showDivider() {
+    if (divider) {
+      return (
+        <Box marginTop="small">
+          <Divider aria-label="divider-drawer" />
+        </Box>
+      );
+    }
+
+    return null;
+  }
+
+  return (
+    <Box width="100%">
+      <Box paddingTop="small" paddingRight="small" paddingLeft="xxlarge">
+        <Heading noPadding role="heading" aria-label="header-drawer">
```

Comment:
Don't define a fixed `aria-label`. You defining a "meaning" that not explain the usage of the "Heading".  If really needed (which i don't think it ) it should be costumizable.